engine/access/index/events_index.go

```diff
@@ -1,45 +1,24 @@
 package index
 
 import (
-	"fmt"
 	"sort"
 
-	"go.uber.org/atomic"
-
 	"github.com/onflow/flow-go/model/flow"
-	"github.com/onflow/flow-go/module/state_synchronization"
-	"github.com/onflow/flow-go/module/state_synchronization/indexer"
 	"github.com/onflow/flow-go/storage"
 )
 
-var _ state_synchronization.IndexReporter = (*EventsIndex)(nil)
-
 // EventsIndex implements a wrapper around `storage.Events` ensuring that needed data has been synced and is available to the client.
-// Note: `EventsIndex` is created with empty report due to the next reasoning:
-// When the index is initially bootstrapped, the indexer needs to load an execution state checkpoint from
-// disk and index all the data. This process can take more than 1 hour on some systems. Consequently, the Initialize
-// pattern is implemented to enable the Access API to start up and serve queries before the index is fully ready. During
-// the initialization phase, all calls to retrieve data from this struct should return indexer.ErrIndexNotInitialized.
-// The caller is responsible for handling this error appropriately for the method.
+// Note: read detail how `SimpleIndex` is working
 type EventsIndex struct {
-	events   storage.Events
-	reporter *atomic.Pointer[state_synchronization.IndexReporter]
+	*SimpleIndex
+	events storage.Events
 }
 
 func NewEventsIndex(events storage.Events) *EventsIndex {
 	return &EventsIndex{
-		events:   events,
-		reporter: atomic.NewPointer[state_synchronization.IndexReporter](nil),
-	}
-}
-
-// Initialize replaces a previously non-initialized reporter. Can be called once.
-// No errors are expected during normal operations.
-func (e *EventsIndex) Initialize(indexReporter state_synchronization.IndexReporter) error {
-	if e.reporter.CompareAndSwap(nil, &indexReporter) {
-		return nil
+		SimpleIndex: NewSimpleIndex(),
```

Comment:
Since this is embedded now, how about instantiating `SimpleIndex` in the access_builder and passing it into the index objects on construction. then, we only need to call `Initialize()` once on the base object